homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,164 @@
+                cover.target_id, cover.callback_device_state
+            )
+            entities.append(cover)
+
+    async_add_entities(entities)
+
+
+class DioChaconCover(DioChaconEntity, CoverEntity):
+    """Object for controlling a Dio Chacon cover."""
+
+    # To avoid having name twice
+    _attr_has_entity_name = False
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    def __init__(
+        self,
+        dio_chacon_client: DIOChaconAPIClient,
+        device: dict[str, Any],
+    ) -> None:
+        """Initialize the cover."""
+
+        _LOGGER.debug(
+            "Adding DIO Chacon SHUTTER Cover with id=%s, name=%s, openlevel=%s, movement=%s and connected=%s",
+            device["id"],
+            device["name"],
+            device["openlevel"],
+            device["movement"],
+            device["connected"],
+        )
+
+        super().__init__(
+            dio_chacon_client, device["id"], device["name"], device["model"]
+        )
+
+        self._attr_available = device["connected"]
+        self._attr_current_cover_position = device["openlevel"]
+        self._attr_is_closing = device["movement"] == ShutterMoveEnum.DOWN.value
+        self._attr_is_opening = device["movement"] == ShutterMoveEnum.UP.value
+        self._attr_is_closed = self._attr_current_cover_position == 0
+
+    # callback coroutine for device state notification pushed from the server
+    def callback_device_state(self, data: dict[str, Any]) -> None:
+        """Receive callback for device state notification pushed from the server."""
+
+        _LOGGER.debug("Data received from server %s", data)
+        self._attr_available = data["connected"]
+        self._attr_current_cover_position = data["openlevel"]
+        self._attr_is_closing = data["movement"] == ShutterMoveEnum.DOWN.value
+        self._attr_is_opening = data["movement"] == ShutterMoveEnum.UP.value
+        self._attr_is_closed = self._attr_current_cover_position == 0
+        self.async_write_ha_state()
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the cover."""
+
+        _LOGGER.debug(
+            "Close cover %s , %s, %s",
+            self.target_id,
+            self._attr_name,
+            self.is_closed,
+        )
+
+        # closes effectively only if cover is not already closing and not fully closed
+        if not self._attr_is_closing and not self.is_closed:
+            self._attr_is_closing = True
+            self.async_write_ha_state()
+
+            await self.dio_chacon_client.move_shutter_direction(
+                self.target_id, ShutterMoveEnum.DOWN
+            )
+
+        # Closed status is effective after the server callback that triggers async_set_updated_data on the coordinator
```

Comment:
The issue I mostly had with the comments in previous feedback was that it was placed at the end which was confusing. But I don't think the comment adds a lot. Maybe a small explanation in the docstring would be better.

Reply:
Corrected.